File: src/main/java/no/nav/pto/veilarbportefolje/domene/TiltakshendelseForBruker.java
package no.nav.pto.veilarbportefolje.domene;

import com.fasterxml.jackson.annotation.JsonCreator;
import no.nav.pto.veilarbportefolje.tiltakshendelse.domain.Tiltakshendelse;
import no.nav.pto.veilarbportefolje.tiltakshendelse.domain.Tiltakstype;
import java.time.LocalDateTime;
import java.util.UUID;

public record TiltakshendelseForBruker(
        UUID id,
        LocalDateTime opprettet,
        String tekst,
        String lenke,
        Tiltakstype tiltakstype) {

    public static TiltakshendelseForBruker of(Tiltakshendelse tiltakshendelse) {
        if(tiltakshendelse != null){
            return new TiltakshendelseForBruker(tiltakshendelse.id(), tiltakshendelse.opprettet(), tiltakshendelse.tekst(), tiltakshendelse.lenke(), tiltakshendelse.tiltakstype());
        } else {
            return null;
        }
    }
    @JsonCreator(mode = JsonCreator.Mode.PROPERTIES)
     public TiltakshendelseForBruker {
     }
}

File: src/main/java/no/nav/pto/veilarbportefolje/domene/frontendmodell/TiltakshendelseForBruker.kt
package no.nav.pto.veilarbportefolje.domene.frontendmodell

import no.nav.pto.veilarbportefolje.tiltakshendelse.domain.Tiltakstype
import java.time.LocalDate
import java.util.*

data class TiltakshendelseForBruker(
    val id: UUID,
    val opprettet: LocalDate,
    val tekst: String,
    val lenke: String,
    val tiltakstype: Tiltakstype
)
